### PR TITLE
Support BlurStyle values in BoxShadow

### PR DESCRIPTION
**Pull request by deepak-penaganti**
`BoxShadow` does not support all the `BlurStyle`s

Adds support for shadow types as mentioned in issue #52999

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 2 new digest(s) on patchset 1.
View them at https://flutter-gold.skia.org/cl/github/86504

**Comment by deepak-penaganti** [comment]
@Piinks @GaryQian we have a clean build and test run. Please do review the code. Thanks

**Review by GaryQian — APPROVED** [review]
This LGTM now! Thank you so much for your contribution!

**Comment by Piinks** [comment]
Gold file changes have to be triaged before this can land, that's why the check is still pending. Please view and approve images using the provided link above @GaryQian .

**Comment by GaryQian** [comment]
@deepak-penaganti I was looking through the goldens generated in https://flutter-gold.skia.org/search?issue=86504&crs=github&patchsets=1&corpus=flutter and it looks like outer, inner, and normal are all generating the same image with the same hash. Are you sure this feature is working as expected?

File: packages/flutter/test/painting/box_painter_test.dart
Comment (GaryQian) [inline packages/flutter/test/painting/box_painter_test.dart:164]: Keep the width consistent, let's use 20 as the other tests use 20

File: packages/flutter/test/painting/box_painter_test.dart
Comment (GaryQian) [inline packages/flutter/test/painting/box_painter_test.dart:162]: Let's also use a wider radius! Simply easier for us to visually inspect the results.

**Comment by deepak-penaganti** [comment]
@GaryQian, Even I am surprised. Below are the visual tests I did on emulator

Outcomes on android emulator
### Normal
image


### Inner
image

### Outer
image

### Solid
image

**Comment by skia-gold** [comment]
Gold has detected about 6 new digest(s) on patchset 2.
View them at https://flutter-gold.skia.org/cl/github/86504

**Comment by deepak-penaganti** [comment]
> @deepak-penaganti I was looking through the goldens generated in https://flutter-gold.skia.org/search?issue=86504&crs=github&patchsets=1&corpus=flutter and it looks like outer, inner, and normal are all generating the same image with the same hash
> image
> Are you sure this feature is working as expected?

It is, I did check before proposing this solution

**Comment by Piinks** [comment]
Oh - I think the golden files look this way because shadows are just rendered as clean black boxes in golden file tests.

**Comment by GaryQian** [comment]
Ahh I see. @Piinks how should we proceed then? Does that mean shadows are not golden testable?

File: packages/flutter/test/painting/box_painter_test.dart
Comment (GaryQian) [inline packages/flutter/test/painting/box_painter_test.dart:149]: Let's add a comment here (as well as before each of the other tests) to indicate that SkiaGold isn't yet able to render shadows in golden tests. Something like:

```
// TODO(garyq): Golden tests currently render shadows as black boxes. We should
// selectively fully render shadows when it is directly relevant.
```

Since the other stuff is tested, we can proceed after adding the TODO comments.

**Comment by Piinks** [comment]
I think there is a way to disable it so the shadow is rendered, it's been a while, but I know we have tests like this. Looking now

**Review by Piinks — COMMENTED** [review]
> how should we proceed then? Does that mean shadows are not golden testable?

They can be, tests like this override it:
https://flutter-gold.skia.org/detail?test=widgets.shadow.PhysicalShape.12&digest=b5f89302c781dcb82dd7f02badc9cfe6

Adding `debugDisableShadows` is what I think is needed here, see here for an example:
https://github.com/flutter/flutter/blob/a5b9b5774aa5eedb8e4944a090e9c65e3de29266/packages/flutter/test/widgets/shadow_test.dart#L137

**Comment by GaryQian** [comment]
Ahh I remember that flag. Well, considering this entire golden *is* a shadow, the the flag kind of makes the test useless. In any case, the other portions are tested and this isn't particularly risky. We should be OK.

**Comment by goderbauer** [comment]
Can we just turn on the flag for this test so we get golden coverage for this?

File: packages/flutter/test/painting/box_painter_test.dart
Comment (goderbauer) [reply packages/flutter/test/painting/box_painter_test.dart:149]: Let's just turn on shadows for this test with the flag mentioned by @Piinks.

**Comment by deepak-penaganti** [comment]
> Can we just turn on the flag for this test so we get golden coverage for this?

How do we add flag?

**Comment by deepak-penaganti** [comment]
> > Can we just turn on the flag for this test so we get golden coverage for this?
> 
> How do we add flag?

Sorry, found the flag. Should I add it.

**Comment by goderbauer** [comment]
> Sorry, found the flag. Should I add it.

Yes, I think so.

**Comment by skia-gold** [comment]
Gold has detected about 8 new digest(s) on patchset 5.
View them at https://flutter-gold.skia.org/cl/github/86504

**Comment by goderbauer** [comment]
@deepak-penaganti Can you rebase this PR to make the checks pass?

**Comment by Piinks** [comment]
I've triaged the golden file images, and I think this should be ready to land, although there is a `flutter_svg` test in the `customer_testing` shard that is failing. @dnfield does this look like the flake from the other day? Or does it look like a real failure?

**Comment by dnfield** [comment]
Yes, this seems to be the same flake. I'm not really clear on why this would happen.

**Comment by Piinks** [comment]
@deepak-penaganti can you try rebasing again? That may resolve the flake. If it does not you may need to open a fresh PR, I ran into this earlier this week.

**Comment by skia-gold** [comment]
Gold has detected about 15 new digest(s) on patchset 5.
View them at https://flutter-gold.skia.org/cl/github/86504

**Pull request closed without merging**
